src/main/java/minicraft/core/io/FileHandler.java

```diff
@@ -132,4 +139,64 @@ public FileVisitResult visitFileFailed(Path p, IOException ex) {
 			deleteFolder(origFolder.toFile());
 	}
 
+	/** https://stackoverflow.com/questions/1429172/how-do-i-list-the-files-inside-a-jar-file/1429275#1429275 */
+	public static ArrayList<String> listResources() {
+		ArrayList<String> names = new ArrayList<>();
+		try {
+			CodeSource src = Game.class.getProtectionDomain().getCodeSource();
+			if (src != null) {
+				URL jar = src.getLocation();
+				ZipInputStream zip = new ZipInputStream(jar.openStream());
+				int reads = 0;
+				while (zip.available() == 1) {
+					ZipEntry e = zip.getNextEntry();
+
+					// e is either null if there are no entries left, or if
+					// we're running this from an ide
+					if (e == null) {
+						if (reads > 0) break;
+						else {
+							return listResourcesUsingIDE();
+						}
+					}
+					reads++;
+					names.add(e.getName());
+				}
+			} else {
+				Logging.RESOURCEHANDLER_LOCALIZATION.error("Failed to get code source.");
+				return names;
+			}
+
+			return names;
+		} catch (IOException e) {
+			CrashHandler.errorHandle(e);
+			return names;
+		}
+	}
+
+	/**
+	 * Gets a list of paths to where the localization files are located on your disk, and adds them to the "localizationFiles" HashMap.
+	 * The path is relative to the "resources" folder.
+	 * Will not work if we are running this from a jar.
+	 */
+	private static ArrayList<String> listResourcesUsingIDE() {
```

Comment:
Check out the changes made by [KalmeMarq](https://github.com/MinicraftPlus/minicraft-plus-revived/commits?author=KalmeMarq).

Reply:
Also, fix the new conflicts in 
src/main/java/minicraft/screen/ResourcePackDisplay.java
src/main/java/minicraft/screen/SkinDisplay.java